### PR TITLE
Added test to create a new dynamic cluster in the domain using online WLST and verify server starts successfully

### PR DESCRIPTION
**Pull request by hzhao-github**
Jenkins jobs

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11629
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11630

Jenkins jobs with latest changes:

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11649
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11650

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        "Failed to write the WLST properties to file");
+    logger.info("WLST property file is: {0} ", wlstPropertiesFile.getAbsolutePath());
+
+    // create a new dynamic cluster using an online WLST script
+    logger.info("Creating a new dynamic cluster using an online WLST script");
+    Path createDynClusterScript = Paths.get(RESOURCE_DIR, "python-scripts", "create-dynamic-cluster.py");
+    executeWLSTScript(createDynClusterScript, wlstPropertiesFile.toPath(), domainNamespace);
+
+    // patch the domain to increase the introspectVersion value to triage the introspector and verify
+    String introspectVersion = assertDoesNotThrow(() -> getNextIntrospectVersion(domainUid, domainNamespace));
```
Comment (alai8) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java:162]: Should we use the patchDomainResourceWithNewIntrospectVersion method in oracle.weblogic.kubernetes.actions.impl.Domain class?

**Comment by alai8** [comment]
LGTM. Just a couple minor comments.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * The test class verifies that a newly added dynamic cluster is not considered as a configured cluster
+ * and server in the newly added dynamic cluster is started successfully.
+ */
+@DisplayName("Verify that server in newly added dynamic cluster is started successfully")
+@IntegrationTest
+@Tag("oke-parallel")
+class ItDynAsConfigCluster {
```
Comment (alai8) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java:51]: class name a bit misleading. we are verifying that newly added dynamic cluster via online WLST is not treated by introspector as a configured cluster. Maybe something like ItAddNewDynamicCluster instead?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java:162]: hmm, a good question. This method is there for a while and used by so many test suites. I will discuss it with Pani and create a new JIRA to move it to DomainUtil.java if he ok it

Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java:162]: and this test suite doesn't use it, patchDomainResourceWithNewIntrospectVersion :(

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+import static oracle.weblogic.kubernetes.utils.PodUtils.getExternalServicePodName;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static oracle.weblogic.kubernetes.utils.WLSTUtils.executeWLSTScript;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * The test class verifies that a newly added dynamic cluster is not considered as a configured cluster
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java]: changed

**Review by alai8 — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDynAsConfigCluster.java]: JIRA https://jira.oraclecorp.com/jira/browse/OWLS-101203 is created to address this issue. Thanks

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * The test class verifies that a new dynamic cluster added by using an online WLST script
+ * is not considered as a configured cluster and server in the newly added dynamic cluster is started successfully.
+ */
+@DisplayName("Verify that server in newly added dynamic cluster is started successfully")
+@IntegrationTest
+@Tag("oke-parallel")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java:50]: add @Tag("kind-parallel") and Tag related to OKD by checking the pom.xml file.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        "Console login validation");
+
+    // create a new dynamic cluster using an online WLST script
+    createNewDynCluster();
+
+    // verify the managed server pod in newly added dynamic cluster comes up
+    checkPodReadyAndServiceExists(newManagedServerPodPrefix + 1, domainUid, domainNamespace);
+  }
+
+  private void createNewDynCluster() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java:128]: Modify method name to createNewDynamicCluster()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java:50]: added

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java (hunk 5 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAddNewDynamicClusterUsingWlst.java:50]: added

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**